tests/client/core/test_middleware.js

```diff
@@ -68,17 +70,51 @@ describe('Prefix Middleware', () => {
     assert.deepEqual(fakeRes.set.firstCall.args, ['vary', []]);
   });
 
-  it('should render a 404 when a URL exception is found', () => {
+  it('should render 404 when a localeURL exception exists', () => {
```

Comment:
Oh, no, so it renders a 404 no matter what. The middleware is hit before the routes are accessed and they are in the config–it makes sense to intercept them right away.

If we wanted to serve routes like `/firefox/downloads/foo` but 404 on other things this patch would need to be more complicated.